src/parpeoptimization/optimizationProblem.cpp

```diff
@@ -54,6 +54,84 @@ void *getLocalOptimumThreadWrapper(void *optimizationProblemVp) {
     return result;
 }
 
+void optimizationProblemGradientCheckMultiEps(OptimizationProblem *problem,
+                                      int numParameterIndicesToCheck
+                                      ) {//perhaps list of epsilons?
+    // set eps (if no default added yet)
+    std::vector<double> multi_eps {1e-1, 1e-3, 1e-5, 1e-7, 1e-9};
+
+    // setting the number of parameters to the minimum of
+    // numParamaterIndicesToCheck and dimension of the problem
+    int numParameters = problem->cost_fun_->numParameters();
+    numParameterIndicesToCheck =
+        std::min(numParameterIndicesToCheck, numParameters);
+    // choose random parameters to check
+    std::vector<int> parameterIndices(numParameters); // ask Daniel
```

Comment:
Ask what? :)

Reply:
ah :D well i wanted to ask where the difference in `std::vector` und `gsl::span` ist und ob es hauptsächlich ist, dass `gsl::span` variable größe hat und deswegen in dem funktionsaufruf benutzt wird.

Reply:
oh and what the three lines below are used for. I do think i remember that in C++ to use anything with rand, one had to create a seed or something alike. Is it that?

Reply:
Regarding the first question, I'll just point you to https://stackoverflow.com/a/45723820
Second: first line fills the array with 0, 1, 2, ..., the next two are what you said.